.github/workflows/tav-command.yml

```
@@ -18,35 +18,66 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 5
     outputs:
-      versions: ${{ steps.generate-matrix.outputs.versions }}
-      modules: ${{ steps.generate-matrix.outputs.modules }}
-      pull-request: ${{ steps.transform.outputs.pull-request }}
+      versions: ${{ steps.transform.outputs.versions }}
+      modules: ${{ steps.transform.outputs.modules }}
+      pull-request: ${{ steps.pr.outputs.number }}
     steps:
-    - name: Validate github comment
-      uses: elastic/apm-pipeline-library/.github/actions/validate-github-comment@current
+      - name: Validate github comment
+        uses: elastic/apm-pipeline-library/.github/actions/validate-github-comment@main
 
-    - id: transform
-      name: Transform comment to the supported matrix
-      uses: actions/github-script@v6
-      with:
-        script: |
-          core.setOutput('pull-request', context.payload.issue.number)
+      - id: pr
+        name: Fetch PR
+        uses: actions/github-script@v6
+        with:
+          script: core.setOutput('number', context.payload.issue.number)
 
-    - uses: actions/checkout@v3
-      with:
-        ref: refs/pull/${{ steps.transform.outputs.pull-request }}/merge
+      - uses: actions/checkout@v3
+        with:
+          ref: refs/pull/${{ steps.pr.outputs.number }}/merge
 
-    ## TODO: use elastic/apm-pipeline-library/.github/actions/version-framework
-    ##       as soon as https://github.com/elastic/apm-pipeline-library/issues/2171 is done.
-    - id: generate-matrix
-      run: .ci/generate-github-tav-matrix.sh
+      - id: transform
+        name: Transform comment to the supported matrix
+        uses: actions/github-script@v6
+        with:
+          script: |
+            async function f() {
+              const fsPromises = require('fs').promises;
+              const data = await fsPromises.readFile('./.ci/tav.json')
+                                 .catch((err) => core.setFailed('Failed to read file', err));
+              return JSON.parse(data.toString());
+            }
+            const matrix = await f()
+            let modules, versions
+            try {
+              versions = matrix.versions
+              modules = matrix.modules
+            } catch (err) {
+              core.setFailed("Error parsing JSON string")
+              return
+            }
```

```suggestion
            const fs = require('fs')

            let modules, versions
            try {
              const matrix = JSON.parse(fs.readFileSync('./.ci/tav.json'))
              versions = matrix.versions
              modules = matrix.modules
            } catch (err) {
              core.setFailed(`Error loading './.ci/tav.json': ${err}`)
              return
            }
```

Could perhaps simplify this block with `fs.readFileSync` and just a single error catch.